From 6d99970259050dc8969d6f2b73bd385ee1940916 Mon Sep 17 00:00:00 2001
From: Jan Vermaete <jan.vermaete@oip.be>
Date: Fri, 21 Aug 2026 14:22:48 +0200
Subject: [PATCH 1/1] Include the CPM.cmake only when needed.

In OpenEmbedded/Yocto there is not cmake/CPM.cmake.
But the recipe will make all dependencies available so the include under
the if/endif will not happen.

Upstream-Status: Pending

Signed-off-by: Jan Vermaete <jan.vermaete@gmail.com>
---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 76462e6..d6b6201 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,8 +9,6 @@ option(CMAKE_POSITION_INDEPENDENT_CODE "Set -fPIC" ON)
 option(ROSX_PYTHON_BINDINGS "Build Python bindings using nanobind" OFF)
 option(ROSX_HAS_JSON "Add JSON converters" ON)
 
-include(cmake/CPM.cmake)
-
 find_package(ament_cmake QUIET)
 
 if (ament_cmake_FOUND)
@@ -63,6 +61,7 @@ if(ROSX_HAS_JSON)
 
     if(NOT RapidJSON_FOUND)
         message(STATUS "Downloading RapidJSON with CPM")
+	include(cmake/CPM.cmake)
         CPMAddPackage(NAME rapidjson
             GIT_TAG v1.1.0
             GITHUB_REPOSITORY "Tencent/rapidjson"
@@ -136,6 +135,7 @@ if(BUILD_BENCHMARKS)
     pkg_check_modules(LZ4 REQUIRED IMPORTED_TARGET liblz4)
     pkg_check_modules(ZSTD REQUIRED IMPORTED_TARGET libzstd)
 
+    include(cmake/CPM.cmake)
     CPMAddPackage(
         NAME mcap
         GIT_TAG releases/cpp/v2.1.3
-- 
2.43.0

